### PR TITLE
[DS-4084] - Relationships should link to Items via REST API

### PR DESCRIPTION
**Pull request by benbosman**
This commit is the first step to support https://jira.duraspace.org/browse/DS-4084
Using the commits in this PR, the HAL links are included in the relationship REST. The leftId and rightId properties have not yet been removed to avoid breaking the Angular UI.
In the next step, we'll implement a PR in Angular to use the HAL links. For testing purposes, it's probably best this PR is merged prior to the Angular PR
Once this PR and the Angular PR are merged, we'll create one additional REST PR to remove the leftId and rightId properties

You can see this PR in action at e.g. https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/relationships/530

### CODE REVIEW

**Comment by paulo-graca** [comment]
The Travis CI build failed. 

But I don't think it's related with this PR. Please check the error message:

```
Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 143.553 sec - in org.dspace.app.rest.IdentifierRestControllerIT
Running org.dspace.app.rest.CommunityRestRepositoryIT


No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

**Comment by tdonohue** [comment]
@benbosman : If you are willing to rebase all `configurable_entities` PRs on the latest code, I've just merged #2286  to that branch.  This PR should hopefully fix the Travis CI oddities for all recent PRs (including this one).

**Comment by benbosman** [comment]
The Travis failure is indeed not due to a problem in this PR. The code in this PR doesn't cause any errors when performing the tests locally.
I will verify whether I can rebase the branch because Raf is on leave

**Comment by benbosman** [comment]
@tdonohue Thank you for the fix to Travis. The build now succeeds as well

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/link/relation/RelationshipHalLinkFactory.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.dspace.app.rest.RestResourceController;
+import org.dspace.app.rest.link.HalLinkFactory;
+import org.dspace.app.rest.model.ItemRest;
+import org.dspace.app.rest.model.hateoas.RelationshipResource;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.Link;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RelationshipHalLinkFactory extends HalLinkFactory<RelationshipResource, RestResourceController> {
```
Comment (tdonohue) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/link/relation/RelationshipHalLinkFactory.java]: This new class needs JavaDocs to describe it.

Comment (benbosman) [reply dspace-spring-rest/src/main/java/org/dspace/app/rest/link/relation/RelationshipHalLinkFactory.java]: The JavaDoc has been included

**Review by tdonohue — APPROVED** [review]
👍 Code looks good. I haven't tested it locally though.

**Review by paulo-graca — APPROVED** [review]
Thank you for adding this!

**Pull request merged**